doc/map-format.md

```diff
@@ -0,0 +1,188 @@
+# Map Format
+
+**What is this?** This document descrcibes the file format for save files of openage.
```

Comment:
If I understand the article correctly, we need to add to the units additionally fields like:
vector, velocity etc (and if we want a recorded file everything as array with timestamps, this can be done later).
What we need now: unique IDs(uint64) for units. (maybe gold/forest too if we have triggers who can create and destroy them)